File: Lv.1/소수 만들기/solution.js
// https://school.programmers.co.kr/learn/courses/30/lessons/12977
function solution(nums) {
  let prime_count = 0;

  for (let i = 0; i < nums.length; i += 1) {
    for (let j = i + 1; j < nums.length; j += 1) {
      for (let k = j + 1; k < nums.length; k += 1) {
        const sum = nums[i] + nums[j] + nums[k];
        const sqrt = ~~Math.sqrt(sum);
        let is_prime = true;
        for (let l = 2; l <= sqrt; l += 1) {
          if (sum % l === 0) {
            is_prime = false;
            break;
          }
        }
        if (is_prime) {
          prime_count += 1;
        }
      }
    }
  }

  return prime_count;
}

File: Lv.1/소수 만들기/solution.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12977?language=python3
import math

def solution(nums):
    answer = 0

    for i, n in enumerate(nums):
        nums2 = nums[i + 1:]
        for j, n2 in enumerate(nums2):
            nums3 = nums2[j + 1:]
            for k, n3 in enumerate(nums3):
                t = n + n2 + n3
                is_prime = True
                for l in range(2, int(t / 2)):
                    if t % l == 0:
                        is_prime = False
                        break
                        
                answer += (1 if is_prime else 0)
                    

    return answer